components/datetime/src/provider/mod.rs

```diff
@@ -6,24 +6,53 @@
 //!
 //! Read more about data providers: [`icu_provider`]
 
+/// Data providers for the Gregorian Calendar.
 pub mod gregory;
+
 pub(crate) mod helpers;
+
+/// Data providers for time zones.
 pub mod time_zones;
 
+/// A collection of [`ResourceKey`](icu_provider::prelude::ResourceKey) structs for DateTime providers.
 pub mod key {
     use icu_provider::{resource_key, ResourceKey};
+
+    /// A [`ResourceKey`](icu_provider::prelude::ResourceKey) to
+    /// [`gregory::DatesV1`](crate::provider::gregory::DatePatternsV1).
```

Comment:
> Thought

I've been thinking a bit about our documentation style. Manually linking these keywords can be very verbose, and a pain. 

> Question

How do you feel about adding doc-only `use` declaration to the top of the file?

e.g.
```rust
#[cfg(doc)]
use {
    crate::provider::{
        gregory::DateSymbolsV1,
        time_zones::{
            ExemplarCitiesV1, MetaZoneGenericNamesLongV1, MetaZoneGenericNamesShortV1,
            MetaZoneSpecificNamesLongV1, MetaZoneSpecificNamesShortV1, TimeZoneFormatsV1,                           
        },  
    },  
    icu_provider::prelude::ResourceKey,
};
```

I think this style would really unclutter our documentation. And it's nice to have all the doc-only dependencies in one spot. I think it's also nice that it gives readers of the file some intuition of related modules, even though they aren't directly included as dependencies when building the code itself.